vllm/distributed/parallel_state.py

```diff
@@ -674,6 +678,37 @@ def model_parallel_is_initialized():
     return (_TP is not None and _PP is not None)
 
 
+TP_STATE_PATCHED = False
+
+
+@contextlib.contextmanager
+def patch_tensor_parallel_group(world_group: Optional[GroupCoordinator],
+                                tp_group: Optional[GroupCoordinator]):
+    """Patch the tp group temporarily until this function ends.
+    It requires the world group to be patched together to keep the integrity.
+    Args:
+        world_group (Optional[GroupCoordinator]): the world group coordinator
+        tp_group (Optional[GroupCoordinator]): the tp group coordinator
+    """
+    global TP_STATE_PATCHED
+    if (not TP_STATE_PATCHED and world_group is not None
+            and tp_group is not None):
```

Comment:
Can you add some comment on these branches? When would these be None? is it possible for one to be None but not the other? (seems we should assert this isn't the case?)

Reply:
Sorry for the confusion.
I've changed here to assert not TP_STATE_PATCHED.
And I've also cleaned up the tp-patching methods so please take a look, thank you!